### PR TITLE
feat: create source with avro row format

### LINKED ISSUE #5045: support avro row format in frontend
### Is your feature request related to a problem? Please describe.

avro format seems deprecated after migrating to rust frontend. 

### Describe the solution you'd like

create source s(...) with (...) row format avro row schema location ...

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

as title

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Connector (sources & sinks)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

follow ways with row format protobuf, here is a demo

```
create materialized source s with ( connector = 'kafka', topic = 'avro_bin', properties.bootstrap.server = '127.0.0.1:29092', scan.startup.mode = 'earliest') row format avro message 'test_student' row schema location 'file://src/source/src/test_data/simple-schema.avsc';
```

## Refer to a related PR or issue link (optional)

close #5045

### CODE REVIEW

**Comment by tabVersion** [comment]
I ran into some character escaping problems while doing the e2e test. Working on it.

**Comment by tabVersion** [comment]
> I ran into some character escaping problems while doing the e2e test. Working on it.

since avro parser's functionality is well tested in https://github.com/risingwavelabs/risingwave/blob/3202a44552fd5bda3ad715962122b947bcbab829/src/source/src/parser/avro_parser.rs#L408-L473 , I think it is acceptable to just test creating source with avro row format rather than querying the results

File: src/source/src/parser/avro_parser.rs [hunk 1]
```diff
@@ -77,6 +79,87 @@ impl AvroParser {
             Err(arvo_schema.err().unwrap())
         }
     }
+
+    pub fn map_to_columns(&self) -> Result<Vec<ColumnDesc>> {
+        if let Schema::Record { fields, .. } = &self.schema {
+            let mut index = 0;
+            let x = fields
```
Comment (shanicky) [inline src/source/src/parser/avro_parser.rs:86]: Maybe we can inline this x?

**Review by shanicky — APPROVED** [review]
generally LGTM

File: src/source/src/parser/avro_parser.rs [hunk 2]
```diff
@@ -180,6 +262,7 @@ pub(crate) fn from_avro_value(column: &SourceColumnDesc, field_value: Value) ->
 
 impl SourceParser for AvroParser {
     fn parse(&self, payload: &[u8], columns: &[SourceColumnDesc]) -> Result<Event> {
+        println!("payload {:?}", payload);
```
Comment (neverchanje) [inline src/source/src/parser/avro_parser.rs:265]: Pls remove the println

File: src/frontend/src/handler/create_source.rs [hunk 3]
```diff
@@ -112,6 +132,20 @@ pub async fn handle_create_source(
                 pk_column_ids: pk_column_ids.into_iter().map(Into::into).collect(),
             }
         }
+        SourceSchema::Avro(avro_schema) => {
+            assert_eq!(columns.len(), 1);
```
Comment (neverchanje) [inline src/frontend/src/handler/create_source.rs:136]: Is this assertion meaningful or only for temporary purposes?

**Review by neverchanje — CHANGES_REQUESTED** [review]

File: src/frontend/src/handler/create_source.rs (hunk 3 above)
Comment (tabVersion) [reply src/frontend/src/handler/create_source.rs:136]: it aligns with create source with pb row format. I think we can remove it after pk is fully supported.

**Review by huangjw806 — COMMENTED** [review]
CI related LGTM!

**Review by neverchanje — APPROVED** [review]

**Pull request merged**